src/serialize.h

```diff
@@ -596,6 +596,53 @@ class LimitedString
 template<typename I>
 BigEndian<I> WrapBigEndian(I& n) { return BigEndian<I>(n); }
 
+/** Formatter to serialize/deserialize vector elements using another formatter
+ *
+ * Example:
+ *   struct X {
+ *     std::vector<uint64_t> v;
+ *     SERIALIZE_METHODS(X, obj) { READWRITE(Using<VectorUsing<VarInt>>(obj.v)); }
+ *   };
+ * will define a struct that contains a vector of uint64_t, which is serialized
+ * as a vector of VarInt-encoded integers.
+ *
+ * V is not required to be an std::vector type. It works for any class that
+ * exposes a value_type, size, reserve, push_back, and const iterators.
+ */
+template<class Formatter>
+struct VectorUsing
```

Comment:
I don't think the `VectorUsing` and `UsingSelf` names for formatters are the best. Would suggest sticking to the `FooFormatter` convention and renaming `VectorUsing` to `VectorFormatter` and `UsingSelf` to `DefaultFormatter`.

Aside from giving more consistent class names, it should also make usages more readable: `Using<VectorUsing<UsingSelf>>(v)` -> `Using<VectorFormatter<DefaultFormatter>>(v)`

Reply:
That's indeed better. Done.